File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    # 이름과 그리움 점수를 딕셔너리로 매핑
    map_ = {name[i]: yearning[i] for i in range(len(name))}
    
    # 각 사진별로 그리움 점수 합산 계산
    result = [
        sum(map_[person] for person in group if person in map_)
        for group in photo
    ]
    
    return result

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.HashMap;
import java.util.Map;
import java.util.stream.IntStream;
import java.util.Arrays;

public class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        Map<String, Integer> map = IntStream.range(0, name.length)
            .collect(HashMap::new, (m, i) -> m.put(name[i], yearning[i]), Map::putAll);

        return Arrays.stream(photo)
            .mapToInt(p -> Arrays.stream(p)
                                 .filter(map::containsKey)
                                 .mapToInt(map::get)
                                 .sum())
            .toArray();
    }
}